main.go

```diff
@@ -449,6 +475,26 @@ func executeSyncCommand(c *cli.Context, state *state.HelmState, helm helmexec.In
 	return state.SyncReleases(helm, values, workers)
 }
 
+func executeTemplateCommand(c *cli.Context, state *state.HelmState, helm helmexec.Interface) []error {
+	if errs := state.SyncRepos(helm); errs != nil && len(errs) > 0 {
+		return errs
+	}
+
+	if errs := state.UpdateDeps(helm); errs != nil && len(errs) > 0 {
+		return errs
+	}
+
+	if c.GlobalString("helm-binary") != "" {
+		helm.SetHelmBinary(c.GlobalString("helm-binary"))
+	}
+
+	args := args.GetArgs(c.String("args"), state)
```

Comment:
You'll notice that unlike the `sync` function, I pass args to `TemplateReleases`, and apply them to `helm` _after_ the fetch has taken place.

This is because if you're using `--set` in your `--args`, they get applied to the `helm fetch` too, which is incorrect and results in an `unknown flag --set` error.

I believe this to be a bug in `--sync` too but I haven't fixed it under this PR